.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,19 @@
+### 1. Why is this change necessary?
+
+
+### 2. What does this change do, exactly?
+
+
+### 3. Describe each step to reproduce the issue or behaviour.
+
+
+### 4. Please link to the relevant issues (if any).
+
+
+### 5. Checklist
+
+- [ ] I have written tests and verified that they fail without my change
+- [ ] I have updated developer-facing release notes if this change affects external developers
+- [ ] I have written or adjusted the documentation according to my changes
+- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
```

Comment:
package and types are annotations and the latter should be enforced by the linter / PHPStan. So I would rewrite this point to something like this

```suggestion
- [ ] I added the correct package annotation to each `src` file (not strictly necessary for tests)
- [ ] This change has code comments where appropriate, especially for non-obvious lines of code
```